src/app/static/src/js/store/git/actions.ts

```diff
@@ -40,4 +31,28 @@ export const actions: ActionTree<GitState, RunnerRootState> & Record<GitAction,
                 })
         }
     },
+
+    async [GitAction.RefreshGit](context) {
+        context.commit(GitMutation.SetGitRefreshing)
+        await api.get('/git/fetch/')
+            .then(({data}) => {
+                const gitBranches = data.data.map(branch => branch.name);
+                context.commit(GitMutation.SetFetchedGit, gitBranches)
+                determineSelectedBranch(context)
+            })
+    },
+}
+
+function determineSelectedBranch(context: ActionContext<GitState, RunnerRootState>) {
+    const {branches} = context.state
+    let {selectedBranch} = context.state
+    if (branches.length && !branches.some(branch => branch === selectedBranch)) {
+        selectedBranch = branches[0]
+    }
+    if (!branches.length) {
+        selectedBranch = ""
+    }
+    if (selectedBranch !== context.state.selectedBranch) {
+        context.dispatch('SelectBranch', selectedBranch)
+    }
 }
```

Comment:
```suggestion
async function determineSelectedBranch(context: ActionContext<GitState, RunnerRootState>) {
    let {selectedBranch, branches} = context.state
    if (branches.length && !branches.some(branch => branch === selectedBranch)) {
        selectedBranch = branches[0]
    }
    if (!branches.length) {
        selectedBranch = ""
    }
    if (selectedBranch !== context.state.selectedBranch) {
        await context.dispatch('SelectBranch', selectedBranch)
    }
}
```

Reply:
My ide shouts at me it I use `let` for any variable that isn't reassigned. I haven't tested but I think BK might reject that as well.

Reply:
Seems weird having to destructor twice, not efficient. Could you try it and see if buildKite complains. It's alright in my IDE.

Reply:
I guess the complaint appeared in buldkite just as you suggested.